libfwupd/fwupd-client.c

```diff
@@ -1058,6 +1058,7 @@ fwupd_client_quit_async(FwupdClient *self,
 			  cancellable,
 			  fwupd_client_quit_cb,
 			  g_steal_pointer(&task));
+	g_clear_object(&priv->proxy);
```

Comment:
i think this needs to be *in* fwupd_client_quit_cb(), i.e. just after the `/* success */`

Reply:
I put it in `fwupd_client_quit_finish` as that seemed more sensible to me.  If you think it needs to be in `fwupd_client_quit_cb` LMK.

Reply:
> as that seemed more sensible to me

Maybe this isn't a concern for the CLI, but I'm worried about what *thread* and GMainContext is unreffing the proxy. I think fwupd_client_quit_cb has a lesser chance of exploding in our future faces.

Reply:
pushed a fix

Reply:
I pushed a nit-fix -- apologies!